commit 18792e64c86dd7e34ba28e4f61faba472b7bf5fc
Author: Chao Yu <chao@kernel.org>
Date:   Thu Oct 6 23:09:28 2022 +0800

    f2fs: support fault injection for f2fs_is_valid_blkaddr()
    
    This patch supports to inject fault into f2fs_is_valid_blkaddr() to
    simulate accessing inconsistent data/meta block addressses from caller.
    
    Usage:
    a) echo 262144 > /sys/fs/f2fs/<dev>/inject_type or
    b) mount -o fault_type=262144 <dev> <mountpoint>
    
    Signed-off-by: Chao Yu <chao@kernel.org>
    Signed-off-by: Jaegeuk Kim <jaegeuk@kernel.org>

diff --git a/Documentation/filesystems/f2fs.rst b/Documentation/filesystems/f2fs.rst
index 17df9a02ccff..b797e8ec96ed 100644
--- a/Documentation/filesystems/f2fs.rst
+++ b/Documentation/filesystems/f2fs.rst
@@ -199,6 +199,7 @@ fault_type=%d		 Support configuring fault injection type, should be
 			 FAULT_SLAB_ALLOC	  0x000008000
 			 FAULT_DQUOT_INIT	  0x000010000
 			 FAULT_LOCK_OP		  0x000020000
+			 FAULT_BLKADDR		  0x000040000
 			 ===================	  ===========
 mode=%s			 Control block allocation mode which supports "adaptive"
 			 and "lfs". In "lfs" mode, there should be no random